cartoframes/context.py

```diff
@@ -234,16 +233,89 @@ def read(self, table_name, limit=None, index='cartodb_id',
         # choose schema (default user - org or standalone - or shared)
         schema = 'public' if not self.is_org else (
             shared_user or self.creds.username())
-        query = 'SELECT * FROM "{schema}"."{table_name}"'.format(
+
+        table_columns = self._get_table_columns(table_name, schema)
+
+        query = self._get_read_query(table_name, schema, table_columns, limit)
+        result = self.copy_client.copyto_stream(query)
+        df = pd.read_csv(result)
+
+        return self._clean_dataframe_from_carto(df, table_columns, decode_geom)
+
+    def _get_read_query(self, table_name, schema, table_columns, limit=None):
+        """Create the read (COPY TO) query
+
+        Args:
+            table_name (str): Name of table in user's CARTO account.
+            schema (str): table schema name in user's CARTO account.
+            table_columns (dict): column names and types from a table.
+            limit (int, optional): Read only `limit` lines from
+                `table_name`. Defaults to ``None``, which reads the full table.
+
+        Returns:
+            str: COPY TO query
+        """
+        query_columns = list(table_columns.keys())
+        query_columns.remove('the_geom_webmercator')
+
+        query = 'SELECT {columns} FROM "{schema}"."{table_name}"'.format(
             table_name=table_name,
-            schema=schema)
+            schema=schema,
+            columns=', '.join(query_columns))
+
         if limit is not None:
             if isinstance(limit, int) and (limit >= 0):
                 query += ' LIMIT {limit}'.format(limit=limit)
             else:
                 raise ValueError("`limit` parameter must an integer >= 0")
 
-        return self.query(query, decode_geom=decode_geom)
+        return 'COPY ({query}) TO stdout WITH (FORMAT csv, HEADER true)'.format(query=query)
+
+    def _clean_dataframe_from_carto(self, df, table_columns, decode_geom=False):
```

Comment:
I don't see a reason for this to be a class member

Reply:
I will move the read related code into Dataset class created by @alrocar in #570 and then, it will make sense (most of the parameters will be class properties)